File: tools/RiderPlugin/src/dotnet/Sparkitect.RiderPlugin/Navigation/SparkitectNavSpikeAction.cs
using JetBrains.Application.DataContext;
using JetBrains.Application.UI.Actions;
using JetBrains.Application.UI.ActionsRevised.Menu;
using JetBrains.TextControl.DataContext;
using JetBrains.Util;

namespace Sparkitect.RiderPlugin.Navigation;

/// <summary>
/// No-op backend action proving the frontend-id routing rail: a matching frontend RiderAnAction with
/// backendActionId "SparkitectNavSpike" reaches this handler over the existing RD protocol, with no
/// rdgen model. Carries no PSI work — only a trace line confirming the backend fired.
/// </summary>
[Action("SparkitectNavSpike", "Sparkitect Nav Spike")]
public sealed class SparkitectNavSpikeAction : IExecutableAction
{
    private static readonly ILogger Logger =
        JetBrains.Util.Logging.Logger.GetLogger(typeof(SparkitectNavSpikeAction));

    public bool Update(IDataContext context, ActionPresentation presentation, DelegateUpdate nextUpdate)
    {
        return context.GetData(TextControlDataConstants.TEXT_CONTROL) != null;
    }

    public void Execute(IDataContext context, DelegateExecute nextExecute)
    {
        Logger.Trace("SparkitectNavSpike backend action fired");
    }
}

File: tools/RiderPlugin/src/rider/main/kotlin/sparkitect/riderplugin/actions/SparkitectNavSpikeAction.kt
package sparkitect.riderplugin.actions

import com.jetbrains.rider.actions.base.RiderAnAction

/**
 * Frontend no-op spike action. The first constructor argument is the backend action id; it MUST match
 * the backend [Action] id "SparkitectNavSpike", which is how the invocation routes over the existing
 * RD protocol to the backend handler without an rdgen model.
 */
class SparkitectNavSpikeAction : RiderAnAction(
    "SparkitectNavSpike",
    "Sparkitect Nav Spike",
    "No-op spike confirming frontend-to-backend action-id routing")
